homeassistant/components/lcn.py

```diff
@@ -0,0 +1,168 @@
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def get_connection(connections, connection_id=None):
+    """Return the connection object from list."""
+    if connection_id is None:
+        connection = connections[0]
+    else:
+        for connection in connections:
+            if connection.connection_id == connection_id:
+                break
+        else:
+            raise ValueError('Unknown connection_id.')
+    return connection
+
+
+def is_address(value):
+    """Validate the given address string.
+
+    Examples for S000M005 at myhome:
+        myhome.s000.m005
+        myhome.s0.m5
+        myhome.0.5    ("m" is implicit if missing)
+
+    Examples for s000g011
+        myhome.0.g11
+        myhome.s0.g11
+    """
+    matcher = PATTERN_ADDRESS.match(value)
+    if matcher:
+        is_group = (matcher.group('type') == 'g')
+        addr = (int(matcher.group('seg_id')),
+                int(matcher.group('id')),
+                is_group)
+        conn_id = matcher.group('conn_id')
+        return addr, conn_id
+    raise vol.error.Invalid('Not a valid address string.')
+
+
+async def async_setup(hass, config):
+    """Set up the LCN component."""
+    import pypck
+    from pypck.connection import PchkConnectionManager
+
+    hass.data[DATA_LCN] = {}
+    hass.data[DATA_LCN][LIB_LCN] = pypck
+
+    conf_connections = config[DOMAIN][CONF_CONNECTIONS]
+    connections = []
+    connection_names = []
+    for conf_connection in conf_connections:
+        # use 'pchk' as default connection_id (or add a numeric suffix if
+        # pchk' is already in use
+        connection_name = conf_connection[CONF_NAME]
+        if connection_name == DEFAULT_NAME:
+            while connection_name in connection_names:
+                suffix = connection_name.strip(DEFAULT_NAME)
+                c_id = 1 if not suffix else int(suffix) + 1
+                connection_name = '{}{:d}'.format(DEFAULT_NAME, c_id)
+        connection_names.append(connection_name)
+
+        settings = {'SK_NUM_TRIES': conf_connection[CONF_SK_NUM_TRIES],
+                    'DIM_MODE': pypck.lcn_defs.OutputPortDimMode[
+                        conf_connection[CONF_DIM_MODE].upper()]}
+
+        connection = PchkConnectionManager(hass.loop,
+                                           conf_connection[CONF_HOST],
+                                           conf_connection[CONF_PORT],
+                                           conf_connection[CONF_USERNAME],
+                                           conf_connection[CONF_PASSWORD],
+                                           settings=settings,
+                                           connection_id=connection_name)
+
+        # establish connection to PCHK server
+        try:
+            await hass.async_create_task(connection.async_connect())
+            connections.append(connection)
+            _LOGGER.info('LCN connected to "{:s}"'.format(connection_name))
+        except TimeoutError:
```

Comment:
How long is the timeout? We're potentially holding up home assistant from starting until this times out, so it shouldn't be too long. Not more than 15-20 seconds.

Reply:
15 seconds are absolutely sufficient. Passed it as argument to `async_connect` method.